docs/integrations/channels.md

```diff
@@ -0,0 +1,209 @@
+However, take care to ensure you run `django.setup()` instead of
+`get_asgi_application()`, if you need any Django ORM or other Django features in
+Strawberry.
+
+## Channel Layers
+
+The Context for Channels integration includes the consumer, which has an
+instance of the channel layer and the consumer's channel name. Here's an example
+of how this can be used in the schema to provide subscriptions to events
+generated by background tasks or the web server, even if these are executed in
+other threads, processes, or even other servers if you are using a Layer backend
+like Redis or RabbitMQ.
+
+To set this up, you'll need to make sure Channel Layers is configured as per the
+[documentation](https://channels.readthedocs.io/en/stable/topics/channel_layers.html).
+
+Then you'll want to add a subscription that accesses the channel layer and joins
+one or more broadcast groups.
+
+Since listening for events and passing them along to the client is a common use case,
+the base consumer provides a high level API for that using a generator pattern.
+
+For example, a basic chat room implementation might look like this:
+
+```python
+
+from typing import AsyncGenerator
+
+from channels.layers import get_channel_layer
+import strawberry
+from strawberry.channels import StrawberryChannelsContext
+from strawberry.types import Info
+
+
+@strawberry.input
+class ChatRoom:
+    id: str
+
+
+@strawberry.type
+class ChatRoomMessage:
+    id: str
+    message: str
+
+
+@strawberry.type
+class Mutation:
+    @strawberry.mutation
+    async def send_chat_message(
+        self,
+        room: ChatRoom,
+        message: str,
+    ):
+        channel_layer = get_channel_layer()
+        await channel_layer.group_send(
+            f"chat-room-{room.id}",
+            {
+                "type": "chat.message",
+                "room_id": room.id,
+                "message": message,
+            }
+        )
+
+
+@strawberry.type
+class Subscription:
+    @strawberry.subscription
+    async def join_chat_rooms(
+        self,
+        info: Info[StrawberryChannelsContext],
+        rooms: list[ChatRoom],
+    ) -> AsyncGenerator[ChatRoomMessage, None]:
+        """Join and subscribe to messages sent to the given rooms."""
+        ws = info.context.request
+        rooms = [f"chat-room-{room.id}" for room in rooms]
+
+        channel_layer = ws.channel_layer
+        for group in channel_layer.groups:
+            await channel_layer.group_send(
+                group,
+                {
+                    "type": "chat.message",
+                    "room_id": group.split("-")[-1],
+                    "message": f"{self.channel_name} just joined the room!",
+                },
+            )
+
+        async for message in ws.channel_listen("chat.message", groups=rooms):
+            yield ChatRoomMessage(
+                id=message["room_id"],
+                message=message["message"],
+            )
+```
+
```

Comment:
```suggestion
Here are the queries that we can run now.
```gql
 subscription fooChat {
  joinChatRooms(rooms: [{roomName: "room1"}, {roomName: "room2"}], user: "foo") {
    roomName
    message
    currentUser
  }
}

subscription barChat {
  joinChatRooms(rooms: [{roomName: "room1"}, {roomName: "room2"}], user: "bar") {
    roomName
    message
    currentUser
  }
}

mutation echo {
  sendChatMessage(message: "hello room 1", room: {roomName: "room1"})
}
```
```

Reply:
I can't wrap them well since the (```suggestion ``\`) is already wrapping it Anyone knows a solution for that?